sequencer/src/block/full_payload/payload.rs

```diff
@@ -281,3 +281,20 @@ impl hotshot_types::traits::block_contents::TestableBlock<SeqTypes> for Payload
         self.len(&self.ns_table) as u64
     }
 }
+
+// `Committable` used only in `reference_tests.rs`
+#[cfg(any(test, feature = "testing"))]
+impl committable::Committable for Payload {
```

Comment:
I don't like adding this implementation solely to test that the implementation is backwards compatible. Can we just pull out the parts of `reference_test` that deal with serialization into a separate function and not do the commitment test for payload?

Reply:
Done in https://github.com/EspressoSystems/espresso-sequencer/pull/1588/commits/321d1772c6cf24db309e1f87614206f61fd13d08 .

Deserialization tests also rely on `Committable`, so I had to tweak them to instead rely on `Eq + Debug`. This works because everything we're testing impls `Eq` and `Debug`.

If we don't wish to impose these trait bounds on our test types then we could instead simulate the `Committable`-based test without the `Committable` trait bound via serialize + hash. But I did not do that.